pkg/collector/corechecks/snmp/internal/report/report_device_metadata.go

```diff
@@ -348,6 +357,77 @@ func buildNetworkTopologyMetadata(deviceID string, store *metadata.Store, interf
 	}
 	return links
 }
+func buildNetworkTopologyMetadataWithCDP(deviceID string, store *metadata.Store, interfaces []metadata.InterfaceMetadata) []metadata.TopologyLinkMetadata {
+	indexes := store.GetColumnIndexes("cdp_remote.interface_id") // using `cdp_remote.interface_id` to get indexes since it's expected to be always present
+	if len(indexes) == 0 {
+		log.Debugf("Unable to build links metadata: no cdp_remote indexes found")
+		return nil
+	}
+	sort.Strings(indexes)
+	var links []metadata.TopologyLinkMetadata
+	for _, strIndex := range indexes {
+		indexElems := strings.Split(strIndex, ".")
+
+		// The cdpCacheEntry index is composed of 2 elements separated by `.`: cdpCacheIfIndex, cdpCacheDeviceIndex
+		if len(indexElems) != 2 {
+			log.Debugf("Expected 2 index elements, but got %d, index=`%s`", len(indexElems), strIndex)
+			continue
+		}
+
+		cdpCacheIfIndex := indexElems[0]
+		cdpCacheDeviceIndex := indexElems[1]
+
+		remoteDeviceAddress := getRemDeviceAddressByCDPRemIndex(store, strIndex)
+
+		resolvedLocalInterfaceID := deviceID + ":" + cdpCacheIfIndex
+
+		// remEntryUniqueID: The combination of cdpCacheIfIndex and cdpCacheDeviceIndex is expected to be unique for each entry in cdpCacheTable
+		remEntryUniqueID := cdpCacheIfIndex + "." + cdpCacheDeviceIndex
+
+		newLink := metadata.TopologyLinkMetadata{
+			ID:         deviceID + ":" + remEntryUniqueID,
+			SourceType: topologyLinkSourceTypeCDP,
+			Remote: &metadata.TopologyLinkSide{
+				Device: &metadata.TopologyLinkDevice{
+					Name:        store.GetColumnAsString("cdp_remote.device_name", strIndex),
+					Description: store.GetColumnAsString("cdp_remote.device_desc", strIndex),
+					ID:          store.GetColumnAsString("cdp_remote.device_id", strIndex),
+					IDType:      "",
+					IPAddress:   getIpAddressFromHexDecimal(remoteDeviceAddress),
+				},
+				Interface: &metadata.TopologyLinkInterface{
+					ID:          store.GetColumnAsString("cdp_remote.interface_id", strIndex),
+					IDType:      metadata.IDTypeInterfaceName,
+					Description: "",
+				},
+			},
+			Local: &metadata.TopologyLinkSide{
+				Interface: &metadata.TopologyLinkInterface{
+					DDID:   resolvedLocalInterfaceID,
+					ID:     "",
+					IDType: "",
+				},
+				Device: &metadata.TopologyLinkDevice{
+					DDID: deviceID,
+				},
+			},
+		}
+		links = append(links, newLink)
+	}
+	return links
+}
+
+func getRemDeviceAddressByCDPRemIndex(store *metadata.Store, strIndex string) string {
+	remoteDeviceAddressType := store.GetColumnAsString("cdp_remote.device_address_type", strIndex)
+	if remoteDeviceAddressType == "1" {
+		return store.GetColumnAsString("cdp_remote.device_address", strIndex)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** I think you should use `store.GetColumnAsByteArray()` that will give you IP in bytes, this will avoid you the need to use `getIpAddressFromHexDecimal()`.

With the IP in bytes, you can use `net.IP(myValueInBytes).String()` to convert it to doted IP format.